app/controllers/datasets_controller.rb

```diff
@@ -27,6 +21,12 @@ def edit
   def update
     @dataset = Dataset.find(params[:id])
     respond_to do |format|
+      # Update the values in the work table
```

Comment:
This part of the code got a bit more complicated than before because now as the user updates a Dataset we are in fact updating two tables: Work (with the title, collection) and Dataset with the specific metadata for the dataset. I think the general idea is solid but we might want to refactor this code in the future (e.g. move it to the model) but l'd rather wait to see if this design sticks.